### ISSUE TITLE
SC2221/SC2222 fires for non-exhaustive wildcard patterns

### ISSUE THREAD

**Issue by jwueller** [koalaman/shellcheck#2969]
### SC2221/SC2222:

```
ShellCheck - shell script analysis tool
version: 0.10.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net
```

- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

foo="--help"
case "$foo" in
    -h|--help)
      echo "usage: $0 [options]"
      exit 0
      ;;

    -*|--*)
      echo "Unknown option: $1"
      exit 1
      ;;

    *)
      echo "arg: $1"
      shift
      ;;
esac
```

#### Here's what shellcheck currently says:
```
In foo.sh line 10:
    -*|--*)
    ^-- SC2221 (warning): This pattern always overrides a later one on line 10.
       ^-^ SC2222 (warning): This pattern never matches because of a previous pattern on line 10.
```

#### Here's what I wanted or expected to see:
No warning.

**Comment by jwueller** [comment]
Sorry, I am an idiot it seems. I assumed the problem was the third case, when the actual problem is in case 2 itself.
